gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/TokenCreationService.java

```diff
@@ -73,4 +64,37 @@ public String createJwtTokenWithoutCredentials(String user) {
             return authenticationService.createTokenAuthentication(user, jwtTokenString).getCredentials();
         }
     }
+
+    public Map<ZosmfService.TokenType, String> createZosmfTokensWithoutCredentials(String user) {
+
+        if (!isZosmfAvailable()) return Collections.emptyMap();
+
+        log.debug("ZOSMF is available and used. Attempt to authenticate with PassTicket");
+        final String passTicket = generatePassTicket(user);
+
+        return zosmfService.authenticate(new UsernamePasswordAuthenticationToken(user, passTicket)).getTokens();
+    }
+
+    private boolean isZosmfAvailable() {
+        try {
+            return providers.isZosfmUsed() && providers.isZosmfAvailable();
+        } catch (AuthenticationServiceException ex) {
+            // Intentionally do nothing. The issue is logged deeper.
+        }
+
+        return false;
+    }
+
+    private String generatePassTicket(String user) {
+        try {
+            log.debug("Generating PassTicket for user: {} and ZOSMF applid: {}", user, zosmfApplId);
+            String passTicket = passTicketService.generate(user, zosmfApplId);
+            log.debug("Generated PassTicket: {}", passTicket);
```

Comment:
I guess it should be at least with a lower level (trace). Keep in the mind that userId above and this pass ticket could be valid for the next 10 mins. Do we really need to log it?